src/evaluate/evaluate.ts

```diff
@@ -175,8 +191,13 @@ export class Evaluator {
     method: string,
     parameters: any[]
   ) {
-    const record = this.context.referenceable(logicalId);
-    const construct = record.primaryValue as any;
+    const record = this.context.reference(logicalId);
+    const construct = record.instance as any;
+    if (!construct || !construct?.[method]) {
```

Comment:
Did you find a case where this constraint was violated? It should have been caught at type resolution.

Reply:
Good point. I think I had added it for type narrowing. Will take another look.

Reply:
Simplified the code without the check.